compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/EnvironmentResolver.java

```diff
@@ -411,6 +411,7 @@ public void visit(BLangWhile whileNode) {
             this.scope = this.symbolEnv;
             this.acceptNode(whileNode.body, this.symbolEnv);
         }
+        this.acceptNode(whileNode.onFailClause, symbolEnv);
```

Comment:
The while stmt's pos includes the on fail clause right? If so, this call should be within the above if block.

Reply:
Updated in b12ab10395f186458914384cb49212a460a18833